File: src/main/java/icu/takeneko/omms/client/data/system/SystemInfo.java
package icu.takeneko.omms.client.data.system;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.ToString;

@Getter
@AllArgsConstructor
@ToString
public class SystemInfo {

    String osName;
    String osVersion;
    String osArch;
    FileSystemInfo fileSystemInfo;
    MemoryInfo memoryInfo;
    NetworkInfo networkInfo;
    ProcessorInfo processorInfo;
    StorageInfo storageInfo;
}

File: src/main/java/icu/takeneko/omms/client/data/system/SystemInfo.kt
package icu.takeneko.omms.client.data.system

data class SystemInfo(
    var osName: String,
    var osVersion: String,
    var osArch: String,
    var fileSystemInfo: FileSystemInfo,
    var memoryInfo: MemoryInfo,
    var networkInfo: NetworkInfo,
    var processorInfo: ProcessorInfo,
    var storageInfo: StorageInfo
)
